src/client/render2d/pixi_renderer.tsx

```diff
@@ -0,0 +1,119 @@
+import * as classNames from 'classnames'
+import { IReactionDisposer } from 'mobx'
+import { observable } from 'mobx'
+import { action } from 'mobx'
+import { autorun } from 'mobx'
+import { observer } from 'mobx-react'
+import { WebGLRenderer } from 'pixi.js'
+import { CanvasRenderer } from 'pixi.js'
+import { Container } from 'pixi.js'
+import { autoDetectRenderer } from 'pixi.js'
+import * as React from 'react'
+import { Component } from 'react'
+import ReactResizeDetector from 'react-resize-detector'
+
+import { Transform } from '../math/transform'
+
+import { pixiObject } from './pixi/rendering'
+import { RendererProps } from './renderer_props'
+import * as style from './style.css'
+
+@observer
+export class PixiRenderer extends Component<RendererProps> {
+  @observable private resolution: Transform = Transform.of()
+  private canvas: HTMLCanvasElement
+  private renderer: WebGLRenderer | CanvasRenderer
+  private stopAutorun: IReactionDisposer
+
+  componentDidMount() {
+    if (!this.canvas) {
```

Comment:
I'd prefer this to error at runtime, because it shouldn't happen. Just use `this.canvas!` below.